### PR TITLE
Map the creation date to datacite publicationYear

### PR DESCRIPTION
**Pull request by jcoyne**
## How was this change tested?

Tested on stage

### CODE REVIEW

File: spec/services/cocina/to_datacite/attributes_spec.rb [hunk 1]
```diff
@@ -92,6 +100,14 @@
                                     ]
                                   }
                                 ],
+                                event: [
```
Comment (arcadiafalcone) [inline spec/services/cocina/to_datacite/attributes_spec.rb:103]: The DataCite publication year is not the creation or publication date entered by the user - it's (our best guess at) the year the digital object is published to the SDR, similar to the autogenerated citation.

File: app/services/cocina/to_datacite/attributes.rb [hunk 2]
```diff
@@ -42,6 +42,12 @@ def mapped_from_cocina
 
       attr :access, :description
 
+      def publication_year
+        creation_event = description.event.find { |event| event.type == 'creation' }
```
Comment (justinlittman) [inline app/services/cocina/to_datacite/attributes.rb:46]: Would like to see a guard against a missing creation event.

Comment (jcoyne) [reply app/services/cocina/to_datacite/attributes.rb:46]: And then what?  This is a required field.  if it's missing here, the API call will fail.

File: spec/services/cocina/to_datacite/attributes_spec.rb (hunk 1 above)
Comment (jcoyne) [reply spec/services/cocina/to_datacite/attributes_spec.rb:103]: So, current year if not embargoed? I believe that's what autogenerated citation uses.

Comment (arcadiafalcone) [reply spec/services/cocina/to_datacite/attributes_spec.rb]: Correct - embargo year if embargoed, otherwise current year.

**Review by justinlittman — APPROVED** [review]

**Pull request merged**